x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -3301,7 +3334,6 @@ public void testInvalidFork() {
         expectError("FROM foo* | FORK (WHERE x>1 | LIMIT 5)", "line 1:13: Fork requires at least two branches");
         expectError("FROM foo* | WHERE x>1 | FORK (WHERE a:\"baz\")", "Fork requires at least two branches");
 
-        expectError("FROM foo* | FORK ( FORK (WHERE x>1) (WHERE y>1)) (WHERE z>1)", "line 1:20: mismatched input 'FORK'");
         expectError("FROM foo* | FORK ( x+1 ) ( WHERE y>2 )", "line 1:20: mismatched input 'x+1'");
         expectError("FROM foo* | FORK ( LIMIT 10 ) ( y+2 )", "line 1:33: mismatched input 'y+2'");
```

Comment:
Do you mind adding a few more scenarios to 

https://github.com/elastic/elasticsearch/blob/7fb130c53b8174f9c1379e1cc32414e86b427fa8/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java#L4256-L4261

 since this change introduces extra popMode that could lead to empty stack exception?

I can imagine following:
```
row f1=1 | keep f1)
row f1=1 | project f1)
```
and possibly others such enrich, mv_expand that has extra popMoe added.